src/index.ts

```
@@ -1,14 +1,31 @@
 //Define class here
-console.log("hello there!");
+const mockUsers = [
+  { id: 0, name: "minoka", age: 31, favoriteColor: "green" },
+  { id: 1, name: "ted", age: 23, favoriteColor: "black" },
+  { id: 2, name: "ron", age: 41, favoriteColor: "blue" },
+  { id: 3, name: "tim", age: 27, favoriteColor: "red" },
+];
 
 class UserAPI {
-  constructor() {}
+  constructor(users) {
+    this.list = users;
+  }
   // these are the methods needed for basic CRUD
-  addUser(name) {
-    console.log(`inside addUser ${name}`);
+  addUser(user) {
+    const newUser = { id: this.list.length, ...user };
+    this.list.push(newUser);
+    return newUser;
   }
   getUserById(id) {
-    console.log(`looking for user with Id of ${id}`);
+    const targetUser = this.list.filter((user) => user.id === id);
+    if (!targetUser[0]) {
+      return "there is no one found with that id";
```

throw exceptions instead of returning string result. look at the documentation for exceptions